### PR TITLE
adding detection for unusual guest activity

### PR DESCRIPTION
**Pull request by ashwin-patil**
Fixes #

## Proposed Changes

  -  Detection looking for unusual guest activity - Inviting external guest users followed by Azure AD powershell signin
  - Ref : [Scary Azure AD Tenant Enumeration… Using Regular B2B Guest Accounts](https://danielchronlund.com/2021/11/18/scary-azure-ad-tenant-enumeration-using-regular-b2b-guest-accounts/)
  -

### CODE REVIEW

File: Detections/MultipleDataSources/UnusualGuestActivity.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+id: acc4c247-aaf7-494b-b5da-17f18863878a
+name: Unusual guest activity - Inviting external guest user followed by Azure AD powershell signin activity
```
Comment (shainw) [inline Detections/MultipleDataSources/UnusualGuestActivity.yaml:2]: Let's shorten the name a bit and match the description a bit closer, remove the part before hyphen and maybe go with "External guest invitation by default guest followed by unusual PowerShell sign in".  Thoughts?

File: Detections/MultipleDataSources/UnusualGuestActivity.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName), tostring(parse_json(tostring(InitiatedBy.app)).displayName))
+  | where  InitiatedByUser has_any ("live.com#", "#EXT#")
+  | extend parsedUser = iff(InitiatedByUser has "live.com#", tostring(split(InitiatedByUser, "#")[1]),tostring(split(InitiatedByUser, "#EXT#")[1])) ,  InvitationTime = TimeGenerated
+  | join ( 
+  SigninLogs 
+  | where UserType == "Guest" and AppDisplayName == "Microsoft Azure PowerShell"
+  | extend SigninTime = TimeGenerated
+  ) on $left.parsedUser == $right.UserPrincipalName
+  | project InvitationTime, SigninTime, InitiatedByUser, OperationName, AppDisplayName , IPAddress, UserType
+  | extend timestamp = InvitationTime, AccountCustomEntity = InitiatedByUser, IPCustomEntity = IPAddress
```
Comment (shainw) [inline Detections/MultipleDataSources/UnusualGuestActivity.yaml:40]: I do not think we need to include these any longer.

File: Detections/MultipleDataSources/UnusualGuestActivity.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+query: |
+  AuditLogs 
+  | where OperationName in ("Invite external user", "Bulk invite users - started (bulk)","Invite external user with reset invitation status")
+  | extend InitiatedByUser = iff(isnotempty(tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName)), 
+    tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName), tostring(parse_json(tostring(InitiatedBy.app)).displayName))
+  | where  InitiatedByUser has_any ("live.com#", "#EXT#")
+  | extend parsedUser = iff(InitiatedByUser has "live.com#", tostring(split(InitiatedByUser, "#")[1]),tostring(split(InitiatedByUser, "#EXT#")[1])) ,  InvitationTime = TimeGenerated
+  | join ( 
+  SigninLogs 
+  | where UserType == "Guest" and AppDisplayName == "Microsoft Azure PowerShell"
```
Comment (shainw) [inline Detections/MultipleDataSources/UnusualGuestActivity.yaml:36]: UserType is causing the KQL validation error:
https://dev.azure.com/azure/Azure-Sentinel/_build/results?buildId=32689&view=logs&j=2bbe2297-1b5b-54f3-6060-67e7b289f6c3&t=293f4a23-e36d-5efa-3a4e-ea46d4ab6d83&l=11

`[xUnit.net 00:00:02.97]     Kqlvalidations.Tests.KqlValidationTests.Validate_DetectionQueries_HaveValidKql(detectionsYamlFileName: "UnusualGuestActivity.yaml") [FAIL]
  X Kqlvalidations.Tests.KqlValidationTests.Validate_DetectionQueries_HaveValidKql(detectionsYamlFileName: "UnusualGuestActivity.yaml") [28ms]
  Error Message:
   Template Id:acc4c247-aaf7-494b-b5da-17f18863878a is not valid in Line:9 col:9 Errors:The name 'UserType' does not refer to any known column, table, variable or function., Code: 'KS142', Severity: 'Error', Location: '660..668',The name 'UserType' does not refer to any known column, table, variable or function., Code: 'KS142', Severity: 'Error', Location: '915..923'
Expected: True
Actual:   False
`

Nuget package may need to be updated or backend schema check for SigninLogs needs to be updated, chat with Preeti, Sara and Amit - 
Test run for /home/vsts/work/1/s/.script/tests/KqlvalidationsTests/bin/Release/netcoreapp3.1/Kqlvalidations.Tests.dll(.NETCoreApp,Version=v3.1)

**Review by shainw — CHANGES_REQUESTED** [review]

File: Detections/MultipleDataSources/UnusualGuestActivity.yaml (hunk 1 above)
Comment (ashwin-patil) [reply Detections/MultipleDataSources/UnusualGuestActivity.yaml:2]: made changes. added explicit Azure AD powershell instead just powershell

File: Detections/MultipleDataSources/UnusualGuestActivity.yaml (hunk 2 above)
Comment (ashwin-patil) [reply Detections/MultipleDataSources/UnusualGuestActivity.yaml:40]: removed columns and mapped to the fields directly

File: Detections/MultipleDataSources/UnusualGuestActivity.yaml (hunk 3 above)
Comment (Amitbergman) [reply Detections/MultipleDataSources/UnusualGuestActivity.yaml:36]: Should be fixed in this PR - https://github.com/Azure/Azure-Sentinel/pull/3561

**Review by Amitbergman — APPROVED** [review]

**Review by shainw — APPROVED** [review]

**Pull request merged**